refocus-bdk-server.js

```diff
@@ -188,46 +204,87 @@ module.exports = (config) => {
 
     const socket = io.connect(connectUrl, opts);
 
-    socket.on(initalizeEventName, (data) => {
+    socket.on(initalizeEventName, async (data) => {
       const eventData = JSON.parse(data);
       const room = eventData[initalizeEventName];
+      const { id, updatedAt } = room;
+      const hasAlreadyBeenConsumed = cache &&
+        await cache.hasBeenConsumed(id, updatedAt, initalizeEventName);
+      if (hasAlreadyBeenConsumed) {
+        return;
+      }
       app.emit('refocus.internal.realtime.bot.namespace.initialize', room);
-      log.realtime('New Room', room);
     });
 
-    socket.on(settingsChangedEventName, (data) => {
+    socket.on(settingsChangedEventName, async (data) => {
       const eventData = JSON.parse(data);
       const room = eventData[settingsChangedEventName];
```

Comment:
is this a room?

Reply:
yeah, this event is fired when the settings of a room are changed eg. it goes from `active` to `inactive`. the event includes the entire room object.